contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -536,21 +666,60 @@ SetColMetadataForTvp(ParameterToken temp, const StringInfo message, uint64_t *of
 			if (i == 1)
 				temp->tvpInfo->tvpTypeSchemaName = tempStringInfo->data;
 			else
+			{
 				temp->tvpInfo->tvpTypeName = tempStringInfo->data;
-
+				temp->tvpInfo->tableName = tempStringInfo->data;
+			}
 		}
 		else if (i == 2)
 		{
-			/* Throw error if TabelType-Name is not provided */
-			ereport(ERROR,
-					(errcode(ERRCODE_PROTOCOL_VIOLATION),
-					 errmsg("The incoming tabular data stream (TDS) remote procedure call (RPC) protocol stream is incorrect. "
-							"Table-valued parameter %d, to a parameterized string has no table type defined.",
-							temp->paramOrdinal + 1)));
+			if (!xactStarted)
+				StartTransactionCommand(); 
+			user_id = GetSessionUserId();
+			target_arg_name = temp->paramMeta.colName.data;
+			curr_db = pltsql_plugin_handler_ptr->get_cur_db_name();
+
+			/* Get procedure namespaceid. */
+			obj_schema_oid = get_proc_namespace_oid(&proc_name,curr_db);
+
+			/* Fetch proargtype value of our targeted variable */
+			tvp_proargtype = tds_get_proargtypes_oid(proc_name, obj_schema_oid, user_id, target_arg_name);
+
+			/* Search in pg_type by object_id */
+			tuple = SearchSysCache1(TYPEOID, ObjectIdGetDatum(tvp_proargtype));
+			/* check if user have right permission on object */
+			if (HeapTupleIsValid(tuple) && object_aclcheck(TypeRelationId, tvp_proargtype, user_id, ACL_USAGE) == ACLCHECK_OK)
+			{
+				Form_pg_type pg_type = (Form_pg_type) GETSTRUCT(tuple);
+				typename = NameStr(pg_type->typname);
+				typnamespace_oid = pg_type->typnamespace;
+				typnamespace = get_namespace_name(typnamespace_oid);
+
+				db_prefix = palloc0(BBF_NAMEDATALEND);
+				snprintf(db_prefix, BBF_NAMEDATALEND , "%s_", curr_db);
+
+				/* Remove db_prefix from begining of typnamespace to get the tvpTypeSchemaName*/
+				if (strncmp(typnamespace, db_prefix, strlen(db_prefix)) == 0) {
+					tvpschemaname = typnamespace + strlen(db_prefix);
+					oldContext = MemoryContextSwitchTo(TopMemoryContext);
+					temp->len += strlen(tvpschemaname);
+					temp->tvpInfo->tvpTypeSchemaName = pstrdup(tvpschemaname);
+					MemoryContextSwitchTo(oldContext);
```

Comment:
Why are we doing this manually? There are already functions for this like get_logical_schema_name. Please check for internal utility functions since most of them should be available

Reply:
Modified. Using get_logical_schema_name now